src/uct/tcp/sockcm/sockcm_iface.c

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2017-2019.  ALL RIGHTS RESERVED.
+ * Copyright (C) NVIDIA Corporation. 2019.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "sockcm_iface.h"
+#include "sockcm_ep.h"
+#include <uct/base/uct_worker.h>
+#include <ucs/sys/string.h>
+#include <ucs/sys/sock.h>
+
+enum uct_sockcm_process_event_flags {
+    UCT_SOCKCM_PROCESS_EVENT_DESTROY_SOCK_ID_FLAG = UCS_BIT(0),
+    UCT_SOCKCM_PROCESS_EVENT_ACK_EVENT_FLAG       = UCS_BIT(1)
+};
+
+static ucs_config_field_t uct_sockcm_iface_config_table[] = {
+    {"BACKLOG", "1024",
+     "Maximum number of pending connections for a (listening?) socket.",
+     ucs_offsetof(uct_sockcm_iface_config_t, backlog), UCS_CONFIG_TYPE_UINT},
+
+    {"SOCK_ID_QUOTA", "64",
+     "How many sockcm connections can progress simultaneously.",
+     ucs_offsetof(uct_sockcm_iface_config_t, sock_id_quota), UCS_CONFIG_TYPE_UINT},
+
+    {NULL}
+};
+
+static UCS_CLASS_DECLARE_DELETE_FUNC(uct_sockcm_iface_t, uct_iface_t);
+
+static ucs_status_t uct_sockcm_iface_query(uct_iface_h tl_iface,
+                                           uct_iface_attr_t *iface_attr)
+{
+    memset(iface_attr, 0, sizeof(uct_iface_attr_t));
+
+    iface_attr->iface_addr_len  = sizeof(ucs_sock_addr_t);
+    iface_attr->device_addr_len = 0;
+    iface_attr->cap.flags       = UCT_IFACE_FLAG_CONNECT_TO_SOCKADDR    |
+                                  UCT_IFACE_FLAG_CB_ASYNC               |
+                                  UCT_IFACE_FLAG_ERRHANDLE_PEER_FAILURE;
+    /* User's private data size is UCT_SOCKCM_UDP_PRIV_DATA_LEN minus room for
+     * the private_data header (to hold the length of the data) */
+    iface_attr->max_conn_priv   = UCT_SOCKCM_MAX_CONN_PRIV;
+
+    return UCS_OK;
+}
+
+static int uct_sockcm_iface_is_reachable(const uct_iface_h tl_iface,
```

Comment:
ucs_empty_function_return_zero
this function should not even be used for client-server type iface

Reply:
@yosefe I've removed this now but I see this function in `rdmacm_iface` as well. In this sense sockcm isn't different right?

Reply:
it's not supposed to be in rdmacm as well. @alinask can you pls fix that?

Reply:
Also, I see that `iface_is_reachable` returns 1 in the case of `rdmacm`.  Should we be using `ucs_empty_function_return_zero` if this is the case?

Reply:
@yosefe removing uct_rdmacm_iface_is_reachable from rdmacm_iface.c will mean to also remove this requirement:  https://github.com/openucx/ucx/blob/master/src/uct/base/uct_iface.c#L392 .
ok?

Reply:
we should set it to ucs_empty_function_return_zero

Reply:
https://github.com/openucx/ucx/pull/3546